modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/PainlessParser.java

```diff
@@ -2834,8 +2837,21 @@ public final LambdaContext lambda() throws RecognitionException {
       }
       setState(448);
       match(ARROW);
-      setState(449);
-      block();
+      setState(451);
+      switch ( getInterpreter().adaptivePredict(_input,38,_ctx) ) {
```

Comment:
yeah, i have no idea about this adaptive prediction, i know we have 36 of these calls today, but no tests are failing (and they fail on the "requires full context" and other stuff like that, that would be super slow). Also unrelated, i have no idea why, in the braces case, the semicolon is still required! Maybe that is related.